### PR TITLE
feat: recoverable `NonLinearPattern`

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Weeder.scala [hunk 1]
```diff
@@ -2135,17 +2138,22 @@ object Weeder {
...
                     // It was unseen until now, so we add it to the seen variables.
                     seen += id.name -> id
                     val l = Name.mkLabel(id)
-                    val patLoc = mkSL(sp11, sp22)
-                    WeededAst.Pattern.Record.RecordLabelPattern(l, p, patLoc).toSuccess
+                    val pat = WeededAst.Pattern.Record.RecordLabelPattern(l, p, patLoc)
+                    pat.toSuccess
+                  case Some(otherIdent) =>
+                    // We recover from a NonLinear pattern by renaming the label.
+                    val l = Name.mkLabel(id)
```
[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Weeder.scala:2153]: Old comment, fixed.

[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Weeder.scala:2153]: We actually just recover by allowing it. Similar to duplicate formals.

**Comment by mlutze** [comment]
What does this error look/feel like when it shows up? I'm not sure I understand the consequences of this recovery

**Comment by magnus-madsen** [comment]
> What does this error look/feel like when it shows up? I'm not sure I understand the consequences of this recovery

It works like shadowing. So you might get redundancy errors with the first `x` is marked as unused.

But in any case, the point is not to have to perfect solution now, it is to make it recoverable so everything else can work.

**Review by mlutze — APPROVED** [review]

**Pull request merged**